bin/SimilarityInterface.py

```diff
@@ -0,0 +1,97 @@
+#notes: many functions are incomplete or disabled because sql isn't implemented yet.
+
+import numpy as np #this does math
+import pandas as pd #this helps with the sorting and the math associated
+import argparse #this helps parse the arguments to the call
+import sys #this is needed to deal with inputs
+
+#sql database to look at
+SQLCommand = "null"
+
+#main function
+def main(args):
+    args = check()
+    simularites = sqlCall(args)  # the table of simularity mesurements
+    sortedSims = sort(simularites)  # the sorted table of simularity mesurements (top = highest)
+    output(sortedSims, args)
+
+# inputs: arguments
+# output: arguments in a better format
+# checks to see if the arguement inputs are correct, if not, program ends
+#temporarly changed because it crashes
+def null_check():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--howmany", help="How many closest simularites to display.", type=int, action="store_true")
+    parser.add_argument("--thresh" , help="threshold for what simularites to display.", type=int, action="store_true")
+    parser.add_argument("--threshArray", help="Array in which to display simularites from. (unclear) (DONT USE THIS UNLESS YOU KNOW WHAT YOU ARE DOING)", action="store_true")
+    parser.add_argument("--ZooID", help="ZooID of the image you want to compare.", action="store_true")
+    parser.add_argument("--UniqueID", help="UniqueID of the image you want to compare.", action="store_true")
+    args = parser.parse_args();
+    #make sure that there arn't too many inputs
+    if (args.howmany and args.tresh):
+        input_error(1)
+    if (args.howmany and args.threshArray):
+        input_error(1)
+    if (args.tresh and args.threshArray):
+        input_error(1)
+    if (args.ZooID and args.UniqueId):
+        input_error(2)
+    if (~(args.ZooID or args.UniqueId)):
+        input_error(3)
+    #this return statment is not completely fleshed out yet
+    return 0;
+#temporary for testing other parts of projet
+def check():
+    return {'count': 7, 'c-type': 'num', 'ID': "memes", 'ID-type': "ZooID"}
+

```

Comment:
So there is a little bit of an easier way to do this.  you could set defaults in the above. parser script. also you can set argument types (i.e. required and not required arguments)
see https://github.com/scottcoughlin2014/X-Pypeline/blob/master/bin/setUpJobs
for an example of argument groups. let me know if you have questions